aiida/engine/processes/workchains/restart.py

```diff
@@ -372,7 +372,7 @@ def get_process_handlers_by_priority(self) -> List[Tuple[int, FunctionType]]:
                 enabled = overrides.pop('enabled', None)
                 priority = overrides.pop('priority', None)
 
-            if enabled is False or not handler.enabled:  # type: ignore[attr-defined]
+            if enabled is False or (enabled is None and not handler.enabled):  # type: ignore[attr-defined]
```

Comment:
```suggestion
            if enabled and handler.enabled:
                handlers.append((priority or handler.priority, handler))
            else:
                continue
```

Is this equal? If so it is more clear. But overall, this is hard to understand in which case the override is happening.

Reply:
No it is not equivalent, because imagine that `handler.enabled` is `False`, meaning the handler is disabled by default, and `enabled` is `True`, meaning the user is overriding it to be enabled, the conditional will evaluate to `False`, causing the handler to be skipped. But it should have been enabled.

What we are testing here is the *if* `enabled` is `True` in the overrides *or* it is not defined at all, but the default of the handler is that it is enabled, then it should be active.

I basically wrote it as the inverse, to know when to skip a handler, i.e., *if* the override is `False` or `override` is not specified and the default is `False` then the handler should not be active.

Not sure how to simplify this further. Maybe assigning intermediate variables would work?
I could write it perhaps as follows:
```python
enabled_override = overrides.pop('enabled', None)
enabled_default = handler.enabled

if enabled_override is False or (enabled_override is None and enabled_default):
    continue
```

Does that help?

Reply:
Variables with the specific name are more clear to me. I thought these two `enable` control two different things. It seems like override's enable control whether do override(I then realise this is meaningless).